urlapi/util.go

```
@@ -0,0 +1,380 @@
+	// if less than around 6 hours missing
+	case diffHeight < 1000:
+		t = getMaxTimeFrom(3)
+	// if less than around 6 hours missing
+	case diffHeight >= 1000:
+		t = getMaxTimeFrom(4)
+	}
+	return time.Now().Add(time.Duration(diffHeight*int64(t)) * time.Millisecond), nil
+}
+func (u *URLAPI) parseProcessInfo(vc *indexertypes.Process,
+	results *types.VochainResults, meta *types.ProcessMetadata) types.APIElectionInfo {
+	var err error
+
+	process := types.APIElectionInfo{
+		Description:        meta.Description["default"],
+		OrganizationID:     vc.EntityID,
+		Header:             meta.Media.Header,
+		ElectionID:         vc.ID,
+		ResultsAggregation: meta.Results.Aggregation,
+		ResultsDisplay:     meta.Results.Display,
+		EndBlock:           vc.EndBlock,
+		StartBlock:         vc.StartBlock,
+		StreamURI:          meta.Media.StreamURI,
+		Title:              meta.Title["default"],
+	}
+
+	// TODO update when blind is added to election
+	// if db.Blind {
+	process.Type = "blind-"
+	// 	} else {
+	// 	resp.Type = "signed-"
+	// }
+	if vc.Mode.EncryptedMetaData {
+		process.Type += "confidential-"
+	} else {
+		process.Type += "plain-"
+	}
+	if vc.Envelope.EncryptedVotes {
+		process.Type += "hidden-results"
+	} else {
+		process.Type += "rolling-results"
+	}
+
+	for _, question := range meta.Questions {
+		newQuestion := types.Question{
+			Title:       question.Title["default"],
+			Description: question.Description["default"],
+		}
+		for _, choice := range question.Choices {
+			newQuestion.Choices = append(newQuestion.Choices, choice.Title["default"])
+		}
+		process.Questions = append(process.Questions, newQuestion)
+	}
+	process.Status = strings.ToTitle(models.ProcessStatus_name[vc.Status])[0:1] +
+		strings.ToLower(models.ProcessStatus_name[vc.Status])[1:]
+
+	if results != nil {
+		process.VoteCount = results.Height
+		if process.Results, err = aggregateResults(meta, results); err != nil {
+			log.Errorf("could not aggregate results: %v", err)
```

IMO this method should return an error and all these log.Errorf and some warnings should return (else it will be hard to find bugs here, the logs are hard to follow sometimes).